### PR TITLE
[ao][sparsity] adding several sparse quantized layers

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81803
* #81802

Summary: This PR adds sparse+quantized versions of the following quantized layers

linear_relu
linear_relu (qat)
linear (qat)

also updated the default sparse+quantized mappings to utilize these new
layers and also updated the qat linear layer to maintain the
sparse_params which also solves the problem for the intrinsic qat layers.

Test Plan: python test/test_ao_sparsity.py TestQuantizedSparseKernels
python test/test_ao_sparsity.py TestQuantizedIntrinsicSparseKernels

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/nn/sparse/intrinsic/qat/modules/linear_relu.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+__all__ = ['SparseQATLinearReLU']
+
+class SparseQATLinearReLU(ao_iqsparsenn.SparseQuantizedLinearReLU):
+    r"""Sparse QAT module (fused).
+
+    Note: There is no fused linear + relu for sparse quantized operation.
+    This module replaces it with sparse_quantized followed by torch.relu.
+
+    Ideally, we would want to inherit from the spars
+    Script doesn't like
```
Comment (supriyar) [inline torch/ao/nn/sparse/intrinsic/qat/modules/linear_relu.py:13]: nit: formatting seems off here

**Review by supriyar — APPROVED** [review]
lgtm, @z-a-f could you take a look as well?

File: torch/ao/nn/sparse/intrinsic/qat/modules/linear_relu.py (hunk 1 above)
Comment (HDCharles) [reply torch/ao/nn/sparse/intrinsic/qat/modules/linear_relu.py]: fixed, good catch

**Review by z-a-f — APPROVED** [review]

**Pull request closed without merging**